scipy/optimize/_linprog.py

```diff
@@ -588,6 +594,19 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
         warning_message = "x0 is used only when method is 'revised simplex'. "
         warn(warning_message, OptimizeWarning)
 
+    # convert sparse to dense array if a simplex method is used and warn
```

Comment:
This logic looks correct now, but again, please do not put it here. You found that there is a function called `_check_sparse_inputs`, and it's purpose is very closely aligned with what you are doing:
>     Check the provided ``A_ub`` and ``A_eq`` matrices conform to the specified
    optional sparsity variables.

It already has logic very similar to what you're adding here:
```
    if meth in dense_methods and sparse_constraint:
        raise ValueError(f"Method '{meth}' does not support sparse "
                         "constraint matrices. Please consider using one of "
                         f"{preferred_methods}.")
```
except that it raises an error instead of performing the conversion.

If we were to leave the existing code in `_check_sparse_inputs`, it would be unreachable because you would have already made `A_eq` and `A_ub` dense. Also, because `_check_sparse_inputs exists, it makes sense to put related code inside that function rather than directly in `linprog`.

So please remove this code from `linprog` and replace the part of `_check_sparse_inputs` that raises an error with your code that performs conversions and raises the warning.